dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/ResourceLink.java

```diff
@@ -0,0 +1,165 @@
+ * `html/portlet/ext/contentlet/field/edit_field.jsp`
+ */
+public class ResourceLink {
+
+    static final String HOST_REQUEST_ATTRIBUTE = "host";
+
+    private static final String HTTP_PREFIX = "http://";
+
+    private static final String HTTPS_PREFIX = "https://";
+
+    private static final String LANG_ID_PARAM = "?language_id=";
+
+    private final String resourceLinkAsString;
+
+    private final String resourceLinkUriAsString;
+
+    private final String mimeType;
+
+    private final FileAsset fileAsset;
+
+    private final boolean editableAsText;
+
+    private final boolean downloadRestricted;
+
+    private ResourceLink(final String resourceLinkAsString, final String resourceLinkUriAsString,
+            final String mimeType,
+            final FileAsset fileAsset,
+            final boolean editableAsText,
+            final boolean downloadRestricted) {
+        this.resourceLinkAsString = resourceLinkAsString;
+        this.resourceLinkUriAsString = resourceLinkUriAsString;
+        this.mimeType = mimeType;
+        this.fileAsset = fileAsset;
+        this.editableAsText = editableAsText;
+        this.downloadRestricted = downloadRestricted;
+    }
+
+    public String getResourceLinkAsString() {
+        return resourceLinkAsString;
+    }
+
+    public String getResourceLinkUriAsString() {
+        return resourceLinkUriAsString;
+    }
+
+    public String getMimeType() {
+        return mimeType;
+    }
+
+    public FileAsset getFileAsset() {
+        return fileAsset;
+    }
+
+    public boolean isEditableAsText() {
+        return editableAsText;
+    }
+
+    public boolean isDownloadRestricted() {
+        return downloadRestricted;
+    }
+
+    @Override
+    public String toString() {
+        return resourceLinkAsString;
+    }
+
+    public static class ResourceLinkBuilder {
+
+        public final ResourceLink build(final HttpServletRequest request, final User user, final Contentlet contentlet) throws DotDataException, DotSecurityException {
+
+            if(!(contentlet.getContentType() instanceof FileAssetContentType)){
+                throw new DotStateException("Can only build Resource Links out of content with type `File Asset`.");
+            }
+
+            final StringBuilder resourceLink = new StringBuilder();
+            final StringBuilder resourceLinkUri = new StringBuilder();
+
+            final Identifier identifier = getIdentifier(contentlet);
+            final Host host = getHost((String)request.getAttribute(HOST_REQUEST_ATTRIBUTE) , user);
+            if (identifier != null && InodeUtils.isSet(identifier.getInode())){
+                if(request.isSecure()){
+                    resourceLink.append(HTTPS_PREFIX);
+                }else{
+                    resourceLink.append(HTTP_PREFIX);
+                }
+                resourceLink.append(host.getHostname());
+                if(request.getServerPort() != 80 && request.getServerPort() != 443){
```

Comment:
it is a good rule I would encapsulate this practice on an Util class, for instance: HttpRequestDataUtil
or something even more decouple from the HttpServletRequest